src/NServiceBus.RabbitMQ.Tests/MessageConverterTests.cs

```diff
@@ -138,7 +137,7 @@ public void Should_override_replyto_header_if_native_replyto_is_present()
         }
 
         [Test]
-        public void TestCanHandleStringHeader()
+        public void TestCanHandleHeadersWithPrimitiveTypes()
```

Comment:
Yeah, those tests names are not good, but I figured it was better to stick with the current convention for now.